protocol/consistencychecks.go

```diff
@@ -45,18 +43,48 @@ func NewCC(savedSTR *DirSTR, useTBs bool, signKey sign.PublicKey) *ConsistencyCh
 	if !useTBs {
 		panic("[coniks] Currently the server is forced to use TBs")
 	}
+	a := NewAuditor(signKey, savedSTR)
 	cc := &ConsistencyChecks{
-		SavedSTR: savedSTR,
+		AudState: a,
 		Bindings: make(map[string][]byte),
 		useTBs:   useTBs,
-		signKey:  signKey,
+		TBs:      nil,
 	}
 	if useTBs {
 		cc.TBs = make(map[string]*TemporaryBinding)
 	}
 	return cc
 }
 
+// CheckEquivocation checks for possible equivocation between
+// an auditors' observed STRs and the client's own view.
+// CheckEquivocation() first verifies the STR range received
+// in msg if msg contains more than 1 STR, and
+// then checks the most recent STR in msg against
+// the cc.verifiedSTR.
+// CheckEquivocation() is called when a client receives a response to a
+// message.AuditingRequest from an auditor.
+func (cc *ConsistencyChecks) CheckEquivocation(msg *Response) error {
+	if err := msg.validate(); err != nil {
+		return err.(ErrorCode)
+	}
+
+	strs := msg.DirectoryResponse.(*STRHistoryRange)
+
+	// verify the hashchain of the received STRs
+	// if we get more than 1 in our range
+	if len(strs.STR) > 1 {
+		if err := cc.verifySTRRange(strs.STR[0], strs.STR[1:]); err != nil {
+			return err
+		}
+	}
+
+	// TODO: if the auditor has returned a more recent STR,
+	// should the client update its savedSTR? Should this
+	// force a new round of monitoring?
+	return cc.checkSTRAgainstVerified(LatestSTRInRange(strs.STR))
```

Comment:
`LatestSTRInRange` is only used here, right? I added this when I using a slice for the `snapshots` instead of a map, but it's no longer necessary now. I'll remove it.